npm/webpack-dev-server-fresh/src/helpers/nextHandler.ts

```diff
@@ -0,0 +1,146 @@
+
+  try {
+    const loadConfigPath = require.resolve('next/dist/server/config', resolvePaths)
+
+    packages.loadConfig = require(loadConfigPath).default
+  } catch (e: any) {
+    throw new Error(`Failed to load "next/dist/server/config" with error: ${e.message ?? e}`)
+  }
+
+  try {
+    const getNextJsBaseWebpackConfigPath = require.resolve('next/dist/build/webpack-config', resolvePaths)
+
+    packages.getNextJsBaseWebpackConfig = require(getNextJsBaseWebpackConfigPath).default
+  } catch (e: any) {
+    throw new Error(`Failed to load "next/dist/build/webpack-config" with error: ${ e.message ?? e}`)
+  }
+
+  return packages
+}
+
+async function loadWebpackConfig ({ devServerConfig, sourceWebpackModulesResult }: PresetHandler): Promise<Configuration> {
+  const { loadConfig, getNextJsBaseWebpackConfig } = getNextJsPackages({ devServerConfig, sourceWebpackModulesResult })
+
+  const nextConfig = await loadConfig('development', devServerConfig.cypressConfig.projectRoot)
+  const runWebpackSpan = await getRunWebpackSpan()
+  const webpackConfig = await getNextJsBaseWebpackConfig(
+    devServerConfig.cypressConfig.projectRoot,
+    {
+      buildId: `@cypress/react-${Math.random().toString()}`,
+      config: nextConfig,
+      dev: true,
+      isServer: false,
+      pagesDir: findPagesDir(devServerConfig.cypressConfig.projectRoot),
+      entrypoints: {},
+      rewrites: { fallback: [], afterFiles: [], beforeFiles: [] },
+      ...runWebpackSpan,
+    },
+  )
+
+  return webpackConfig
+}
+
+export function checkSWC (
+  webpackConfig: Configuration,
+  cypressConfig: Cypress.PluginConfigOptions,
+) {
+  const hasSWCLoader = webpackConfig.module?.rules?.some((rule) => {
+    return typeof rule !== 'string' && rule.oneOf?.some(
+      (oneOf) => (oneOf.use as any)?.loader === 'next-swc-loader',
+    )
+  })
+
+  // "resolvedNodePath" is only set when using the user's Node.js, which is required to compile Next.js with SWC optimizations
+  // If it is not set, they have either explicitly set "nodeVersion" to "bundled" or are are using Cypress < 9.0.0 where it was set to "bundled" by default
+  if (hasSWCLoader && cypressConfig.nodeVersion === 'bundled') {
+    throw new Error(`Cypress cannot compile your Next.js application when "nodeVersion" is set to "bundled". Please remove this option from your Cypress configuration file.`)
+  }
+
+  return false
+}
+
+const existsSync = (file: string) => {
+  try {
+    fs.accessSync(file, fs.constants.F_OK)
+
+    return true
+  } catch (_) {
+    return false
+  }
+}
+
+/**
+ * Next allows the `pages` directory to be located at either
+ * `${projectRoot}/pages` or `${projectRoot}/src/pages`.
+ * If neither is found, return projectRoot
+ */
+export function findPagesDir (projectRoot: string) {
+  // prioritize ./pages over ./src/pages
+  let pagesDir = path.join(projectRoot, 'pages')
+
+  if (existsSync(pagesDir)) {
+    return pagesDir
+  }
+
+  pagesDir = path.join(projectRoot, 'src/pages')
```

Comment:
Should it be `path.join(projectRoot, 'src', 'pages')` to use the correct separator on windows?